src/jgikbase/idmapping/userlookup/jgi_user_lookup_prototype.py

```diff
@@ -0,0 +1,61 @@
+from jgikbase.idmapping.core.user_lookup import UserLookup  # pragma: no cover
+from jgikbase.idmapping.core.user import AuthsourceID, User, Username  # pragma: no cover
+from jgikbase.idmapping.core.tokens import Token  # pragma: no cover
+from typing import Optional, Tuple  # pragma: no cover
+from jgikbase.idmapping.core.arg_check import not_none  # pragma: no cover
+import requests  # pragma: no cover
+from jgikbase.idmapping.core.errors import InvalidTokenError  # pragma: no cover
+from typing import Dict
+
+
+# This is a prototype for interfacing with the JGI authentication system, Caliban.
+# Currently it has no automated tests.
+# The reasons it is a prototype are:
+# 1) JGI uses integer user IDs rather than user names as immutable IDs, which are not very
+#    readable. Maybe this is ok and the UI should be responsible for displaying readable
+#    user names, but it needs some discussion / thought.
+# 2) JGI accounts can be merged, which means a single account can have multiple user IDs.
+#    The prototype does not account for that at this point and as such if a user merges one
+#    account into another, they will lose access to the data linked to the first account.
+#    get_user and is_valid_user should reject merged user accounts.
+# 3) There is no way to specify which users are system admins. This probably isn't possible
```

Comment:
Maybe you could keep a system_admin flag in mongo for users and set it to true manually for certain people.

Reply:
The point of this was that admins for that org could determine who as an id mapper admin without the sysadmins running the id mapper service having to do anything. If that's the solution you might as well just give the user a local account.